kolibri/core/assets/src/views/k-checkbox/index.vue

```diff
@@ -0,0 +1,191 @@
+<template>
+
+  <div
+    class="k-checkbox-container"
+    :class="{ 'k-checkbox-disabled': disabled }"
+    @click.prevent="toggleCheck"
+  >
+    <div class="tr">
+
+      <div class="k-checkbox" :class="{ 'k-checkbox-active': isActive }">
+        <input
+          type="checkbox"
+          class="k-checkbox-input"
+          :name="name"
+          :checked="isCurrentlyChecked"
+          :indeterminate.prop="isCurrentlyIndeterminate"
+          :disabled="disabled"
+          @click.stop="toggleCheck"
+          @focus="isActive = true"
+          @blur="isActive = false"
+        >
+
+        <mat-svg
+          v-if="isCurrentlyIndeterminate"
+          category="toggle"
+          name="indeterminate_check_box"
+          class="k-checkbox-indeterminate"
+        />
+        <mat-svg
+          v-else-if="!isCurrentlyIndeterminate && isCurrentlyChecked"
+          category="toggle"
+          name="check_box"
+          class="k-checkbox-checked"
+        />
+        <mat-svg
+          v-else
+          category="toggle"
+          name="check_box_outline_blank"
+          class="k-checkbox-unchecked"
+        />
+
+      </div>
+
+      <label v-if="label" :for="name" class="k-checkbox-label">{{ label }}</label>
```

Comment:
two issues here:

* leaving `for` optional will break a11y if the dev doesn't provide a name, so we should ensure that the label _always_ gets associated with the checkbox
* `for` actually associates with an `id`, not `name` on the checkbox

---

recommend:

Add a new computed prop to generate a unique ID for the checkbox input:

```
      uniqueCheckboxId() {
        return 'k-checkbox-' + this._uid;
      },
```

set both the label's `:for` attribute and the checkbox's `:id` attribute to `uniqueCheckboxId`